azurerm/resource_arm_policy_set_definition.go

```diff
@@ -0,0 +1,283 @@
+		var policyDefinitions []policy.DefinitionReference
+		err := json.Unmarshal([]byte(policyDefinitionsString), &policyDefinitions)
+		if err != nil {
+			return fmt.Errorf("unable to expand parameters json: %s", err)
+		}
+		properties.PolicyDefinitions = &policyDefinitions
+	}
+
+	definition := policy.SetDefinition{
+		Name:                    utils.String(name),
+		SetDefinitionProperties: &properties,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, name, definition); err != nil {
+		return err
+	}
+
+	// Policy Definitions are eventually consistent; wait for them to stabilize
+	log.Printf("[DEBUG] Waiting for Policy Set Definition %q to become available", name)
+	stateConf := &resource.StateChangeConf{
+		Pending:                   []string{"404"},
+		Target:                    []string{"200"},
+		Refresh:                   policySetDefinitionRefreshFunc(ctx, client, name),
+		Timeout:                   5 * time.Minute,
+		MinTimeout:                10 * time.Second,
+		ContinuousTargetOccurence: 10,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Policy Set Definition %q to become available: %s", name, err)
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmPolicySetDefinitionRead(d, meta)
+}
+
+func resourceArmPolicySetDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).policySetDefinitionsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name, err := parsePolicySetDefinitionNameFromId(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Policy Definition %q - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Policy Definition %+v", err)
+	}
+
+	d.Set("name", resp.Name)
+
+	if props := resp.SetDefinitionProperties; props != nil {
+		d.Set("policy_type", props.PolicyType)
+		d.Set("display_name", props.DisplayName)
+		d.Set("description", props.Description)
+
+		if metadata := props.Metadata; metadata != nil {
+			metadataVal := metadata.(map[string]interface{})
+			metadataStr, err := structure.FlattenJsonToString(metadataVal)
+			if err != nil {
+				return fmt.Errorf("unable to flatten JSON for `metadata`: %s", err)
+			}
+
+			d.Set("metadata", metadataStr)
+		}
+
+		if parameters := props.Parameters; parameters != nil {
+			paramsVal := props.Parameters.(map[string]interface{})
+			parametersStr, err := structure.FlattenJsonToString(paramsVal)
+			if err != nil {
+				return fmt.Errorf("unable to flatten JSON for `parameters`: %s", err)
+			}
+
+			d.Set("parameters", parametersStr)
+		}
+
+		if policyDefinitions := props.PolicyDefinitions; policyDefinitions != nil {
+			policyDefinitionsRes, err := json.Marshal(props.PolicyDefinitions)
```

Comment:
_minor_ we can use the local variable here:

```suggestion
			policyDefinitionsRes, err := json.Marshal(policyDefinitions)
```